### PR TITLE
New: UI & Utilization for Featured Image Generation [ED-15052]

### PR DESCRIPTION
**Pull request by DennisNerush**

### CODE REVIEW

File: modules/ai/assets/js/editor/pages/form-media/views/generate/index.js [hunk 1]
```diff
@@ -153,41 +155,50 @@ const Generate = () => {
...
-												<ImageActions.EditIcon onClick={ () => edit( suggestedPrompt ) } />
-											</ImageActions>
-										</Gallery.Image>
-									) )
-								}
-							</Gallery>
-						</>
 					)
+						: (
+							<>
```
Comment (DennisNerush) [inline modules/ai/assets/js/editor/pages/form-media/views/generate/index.js:160]: Added ```{ predefinedPrompt ? <ImagesPlaceholder />``` as another option to present images placeholders

File: modules/ai/assets/js/gutenberg/excerpt.js [hunk 2]
```diff
@@ -5,50 +5,12 @@ import useUserInfo from '../editor/hooks/use-user-info';
...
 import { RequestIdsProvider } from '../editor/context/requests-ids';
-import styled from 'styled-components';
 import { __ } from '@wordpress/i18n';
+import { useGutenbergPostText } from './post-text-utils';
+import { AiLink, Icon } from './styles';
 
 const { useSelect, useDispatch } = wp.data;
 
-// Function to extract all textual content from a Gutenberg post
-const useGutenbergPostText = ( ) => {
```
Comment (DennisNerush) [inline modules/ai/assets/js/gutenberg/excerpt.js:14]: extracted to a dedicated file

File: modules/ai/assets/js/gutenberg/excerpt.js [hunk 3]
```diff
@@ -120,23 +82,6 @@ AIExcerpt.propTypes = {
 	onClose: PropTypes.func.isRequired,
 };
 
-const Icon = styled.i`
```
Comment (DennisNerush) [inline modules/ai/assets/js/gutenberg/excerpt.js:123]: same - extracted to a dedicated file

File: modules/ai/assets/js/gutenberg/featured-image.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import { AIMediaGenerateApp } from '../media-library/componenets';
+import { useGutenbergPostText } from './post-text-utils';
+import { AiLink, Icon } from './styles';
+import { useState } from '@wordpress/element';
+import useFeaturedImagePrompt from '../editor/hooks/use-featured-image-prompt';
+
+const { useDispatch } = wp.data;
+
+const GenerateFeaturedImageWithAI = () => {
+	const FEATURED_IMAGE_RATIO = '4:3';
```
Comment (netanelavr) [inline modules/ai/assets/js/gutenberg/featured-image.js:12]: why do we need to override the default 1:1 ratio?

**Review by netanelavr — DISMISSED** [review]

File: modules/ai/assets/js/gutenberg/featured-image.js (hunk 4 above)
Comment (DennisNerush) [reply modules/ai/assets/js/gutenberg/featured-image.js:12]: Featured Image shouldn't be square and not a rectangle

**Review by matipojo — DISMISSED** [review]

File: modules/ai/assets/js/media-library/generate.js [hunk 5]
```diff
@@ -48,7 +48,7 @@ const AIMediaGenerateAppWrapper = () => {
 					<Icon className={ 'eicon-ai' } />
 					{ __( 'Generate with Elementor AI', 'elementor' ) }
 				</StyledButton>
-				{ isOpen && <AIMediaGenerateApp onClose={ handleClose } /> }
+				{ isOpen && <AIMediaGenerateApp onClose={ handleClose } setControlValue={ () => {} } /> }
```
[the remark this answers is not in the document]
Comment (DennisNerush) [reply modules/ai/assets/js/media-library/generate.js:51]: it's the flow of the media library so there's no control in that case. Feature image does iniate it

**Review by matipojo — APPROVED** [review]

**Pull request merged**